### ISSUE TITLE
RP2350 Support

### ISSUE THREAD

**Issue by mcknly** [mcknly/breadboard-os#24]
Raspberry Pi has released the RP2350 (Pico 2), which should have the same SDK compatibility making code written for RP2040 portable to RP2350. However, there are several initial challenges blocking direct implementation:
- There are new `PICO_BOARD` and `PICO_PLATFORM` variables for pico2/rp2350 that need to be passed
- Currently the `hardware/rp2040` directory name is tied directly to `PICO_BOARD` definition
- FreeRTOS does not natively support the RP2350 yet (see [here](https://forums.freertos.org/t/freertos-on-pico-2350/21305))

**Comment by mcknly** [comment]
See [this commit](https://github.com/raspberrypi/FreeRTOS-Kernel/commit/4f7299d6ea746b27a9dd19e87af568e34bd65b15) on Raspberry Pi's [FreeRTOS fork](https://github.com/raspberrypi/FreeRTOS-Kernel) for the "unofficial" port for RP2350

**Comment by mcknly** [comment]
To use Raspberry Pi's unofficial port - merge the RP2350-specific changes into an existing copy of FreeRTOS kernel source, open a terminal in the kernel source directory and run:
```
git remote add raspberrypi https://github.com/raspberrypi/FreeRTOS-Kernel.git
git checkout -b rp2350
git pull raspberrypi 4f7299d6ea746b27a9dd19e87af568e34bd65b15
```
This will merge Raspberry Pi's changes. To use the RP2350 port, `portable/ThirdParty/GCC/RP2350_ARM_NTZ/FreeRTOS_Kernel_import.cmake` is sourced, and the `PICO_PLATFORM` variable must be set to "rp2350" to get it to work.

**Comment by mcknly** [comment]
Additional config options are necessary in `FreeRTOSConfig.h` to support ARMv8m (RP2350 is now Cortex-m33 vs ARMv7m/Cortex-m0+ on RP2040) - see [this](https://www.freertos.org/Community/Blogs/2020/using-freertos-on-armv8-m-microcontrollers) post

**Comment by mcknly** [comment]
`FreeRTOSConfig.h` with RP2350-specific configurations from Raspberry Pi in [pico-examples](https://github.com/raspberrypi/pico-examples/commit/7fe60d6b4027771e45d97f207532c41b1d8c5418#diff-6c224fc90880bc0c01f8788622ac6fdbe602b85ecf795996a7df9228c4898139) - note `configMAX_SYSCALL_INTERRUPT_PRIORITY` option is also necessary

**Comment by mcknly** [comment]
Merged into v0.3-pre (#26) to be part of 0.3 release
